stages/org.osbuild.dnf.config

```diff
@@ -60,13 +78,43 @@ def configure_variable(tree, name, value):
     with open(f"{tree}{vars_directory}/{name}", "w") as f:
         f.write(value + "\n")
 
+def make_value(value):
+    val = str(value)
+    return val
+
+def make_section(cfg, name, settings):
+    if not cfg.has_section(name):
+        cfg.add_section(name)
+
+    for key, value in settings.items():
+        val = make_value(value)
+        cfg.set(name, key, val)
+
+def make_dnf_conf(tree, config_options):
+    """
+    Merges the given config object into /etc/dnf/dnf.conf, overwriting existing
+    values.
+    """
+    dnfconf = f"{tree}/etc/dnf/dnf.conf"
+
+    cfg = configparser.ConfigParser()
+    cfg.read(dnfconf)
+
+    for section, items  in config_options.items():
+        make_section(cfg, section, items)
+
+    with open(dnfconf, "w") as f:
+        os.fchmod(f.fileno(), 0o644)
+        cfg.write(f, space_around_delimiters=False)
 
 def main(tree, options):
     variables = options.get("variables", [])
 
     for variable in variables:
         configure_variable(tree, variable["name"], variable["value"])
 
+    make_dnf_conf(tree, options.get("config", {}))
```

Comment:
I wonder if we should not do anything if the config is empty, because currently we will read and write the file, changing its timestamp.